packages/nextjs/context/nvm/NvmContext.tsx

```diff
@@ -0,0 +1,162 @@
+
+  // at connector update, get the provider, initialize nvm and login
+  useEffect(() => {
+    const startNvm = async () => {
+      try {
+        await getProvider();
+        await initSdk();
+        await login();
+        await publishAsset();
+      } catch (error) {
+        console.error(error);
+      }
+    };
+
+    startNvm();
+    // getProvider()
+    // initSdk();
+    // login();
+    // publishAsset(); //That is just for testing. this function should be used in the Confirm component
+  }, [wagmiConfig.connector]);
+
+  const getProvider = async () => {
+    try {
+      const provider = await connector?.getProvider();
+      setState(prevState => ({ ...prevState, provider }));
+    } catch (error) {
+      console.error;
+    }
+  };
+
+  const initSdk = async () => {
+    debugger
+    if (!signer || !state.provider) return;
+    const config: NeverminedOptions = {
+      // The web3 endpoint of the blockchain network to connect to, could be an Infura endpoint, Quicknode, or any other web3 provider
+      web3ProviderUri: "https://goerli-rollup.arbitrum.io/rpc",
+      web3Provider: state.provider,
+      // web3Provider: provider,
+      // The url of the marketplace api if you connect to one. It could be your own service if you run a Marketplace API
+      marketplaceUri: "https://marketplace-api.goerli.nevermined.app",
+      // The url to a Nevermined node. It could be your own if you run a Nevermined Node
+      neverminedNodeUri: "https://node.goerli.nevermined.app",
+      // The public address of the above Node
+      neverminedNodeAddress: "0x5838B5512cF9f12FE9f2beccB20eb47211F9B0bc",
+      // The url to access the nevermined subgraphs required to query for on-chain events
+      graphHttpUri: "https://api.thegraph.com/subgraphs/name/nevermined-io/public",
+      // Folder where are copied the ABIs of the Nevermined Smart Contracts
+      artifactsFolder: "http://localhost:3000/artifacts",
+      marketplaceAuthToken:
+        localStorage.getItem("marketplaceAuthToken") != null
+          ? (localStorage.getItem("marketplaceAuthToken") as string)
+          : "",
+    };
+    setNvmConfig(config);
+    try {
+      const sdk: Nevermined = await Nevermined.getInstance(config);
+      setNvm(sdk);
+      console.log(await sdk.utils.versions.get());
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  const login = async () => {
+    try {
+      if (!nevermined) return;
+
+      const publisher = nevermined.accounts.getAccount(address as string);
+
+      const clientAssertion = await nevermined.utils.jwt.generateClientAssertion(publisher);
+      const loginResult = await nevermined.services.marketplace.login(clientAssertion);
+      localStorage.setItem("marketplaceAuthToken", loginResult);
+      const payload = decodeJwt(loginResult);
+      debugger
+      setState(prevState => ({ ...prevState, payload, publisher }));
+    } catch (error) {
+      console.error;
```

Comment:
fix: actually log the error

```suggestion
      console.error(error);
```